### DISCUSSION TITLE
rAudio 20230609 NAS behaviour change

### DISCUSSION THREAD

**Discussion by markblinkhorn** [rern/rAudio#1346] [Bug Reports]
CIFS NAS share no longer shows context menu at first folder level. This is a bit tricky to explain, but here goes: example folder structure is /mnt/MPD/NAS/Music/ArtistName which appears under the Library as NAS/Music/ArtistName - click on the Folder icon for ArtistName and nothing happens. Click on the ArtistName to open the folder and the contents of the folder are shown - this might be further folders or individual files - click on the icon for any contained item and the context menu appears (Add, Play, Replace, etc). Earlier versions of rAudio did display the context menu at the previous folder level. I have a number of folders with files at the first level and can no longer add them to playlists because of the lack of the context menu. I could move the files into deeper folders (and rebuild the database) but there are very many of them and it would be rather tiresome.

I can't confirm/deny this happens with NFS shares as I don't have any set up.

**Comment by ceejayemm** [comment]
Seems to be doing the same, intermittently, on a NAS NFS music library too, after updating to 20230609.

Chris

**Comment by markblinkhorn** [comment]
20230620 stil misbehaves in the same way. I was at a bit of a loose end today so, just to see what happens, I did a clean install of 20230521 and restored a backup from a running 20230620. Library worked fine - context menu appears on every item (folder or file) in the library. I then applied 20230620 update from the Addon menu and the problem is back again - context menu is broken.

As a work round, I'm going to re-install a clean 20230521 and leave it at that for now.

For Info - this works...

# cat /boot/cmdline.txt
root=PARTUUID=a76cc5df-02 rw rootwait selinux=0 plymouth.enable=0 smsc95xx.turbo_mode=N dwc_otg.lpm_enable=0 ipv6.disable=1 fsck.repair=yes isolcpus=3 console=tty3 quiet loglevel=0 logo.nologo vt.global_cursor_default=0

# cat /boot/config.txt
initramfs initramfs-linux.img followkernel
disable_overscan=1
disable_splash=1
dtparam=audio=on

# pacman -Qs 'firmware|bootloader' | grep ^local | cut -d/ -f2
firmware-raspberrypi 20230125-1
linux-firmware 20230404.2e92a49f-1
linux-firmware-whence 20230404.2e92a49f-1
raspberrypi-bootloader 20230517-1
raspberrypi-firmware 20230516-1

# /etc/modules-load.d/raspberrypi.conf
snd-bcm2835
rAudio 20230310

**Comment by markblinkhorn** [comment]
@rern - Thanks for all the great work! I'd appreciate your comments on this hack.

I have finally tracked down the source of my issue and can report the following workaround has resolved it with 3 different rPi installations - an rPi1 (really!), rPi2B and rPi3B+

Here's what I did, on an installation of  rAudio v20230630:

- nano /srv/http/assets/js/function.js
- jump to line 1229 (in function renderLibraryList)
- insert the following code:


`        $( '#lib-mode-list' ).after( html ).promise().done( () => { `
`                if ( $( '.licover' ).length ) { // V.librarytrack`
`                        V.librarytrack = true;`
`                        if ( $( '#liimg' ).attr( 'src' ).slice( 0, 16 ) === '/data/shm/online' ) $( '.licoverimg ' ).append( V.icoversave );`
`                } else {`
`                        V.librarytrack = false;`
`                        imageLoad( 'lib-list' );`
`                }`

- delete out the following (now redundant) code:

`        $( '#lib-mode-list' ).after( html ).promise().done( () => {`
`              V.librarytrack = $( '#lib-list .li-icon' ).hasClass( 'i-music' );`
`              if ( $( '.licover' ).length ) { // V.librarytrack`
`                      if ( $( '#liimg' ).attr( 'src' ).slice( 0, 16 ) === '/data/shm/online' ) $( '.licoverimg ' ).append( V.icoversave );`
`              } else {`
`                      imageLoad( 'lib-list' );`
`                }`

- refresh the browser (ctrl-F5) to restore the missing context menu functionality.

This effectively replaces the 20230630 code fragment with that from 20230521. I haven't (yet) noticed any newly created issues from doing this but it does fix my immediate problem.

Mark

**Reply by rern** [reply]
Thank you for the code.

Please revert back and try replacing the line:
`V.librarytrack = $( '#lib-list .li-icon' ).hasClass( 'i-music' );`
to
`V.librarytrack = ! $( '#lib-list li:not( .licover )' ).find( '.i-folder, .iconthumb' ).length;`

Then command: `cacheBust`

It has to take the enabled `Hide coverart band` option, which `.licover` won't exist, as well.

**Comment by markblinkhorn** [comment]
@rern - I can confirm that works. Thanks for sorting.

Mark
